File: Easy/roman_to_integer.go
package main

import (
	"fmt"
	"strings"
)

func romanToInt(s string) int {
	symbolMap := map[string]int{"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
	total := 0
	strSplit := strings.Split(strings.ToUpper(s), "")

	for i, c := range strSplit {
		if _, ok := symbolMap[c]; !ok {
			break
		}

		if i > 0 {
			if (c == "V" || c == "X") && strSplit[i-1] == "I" {
				total -= 2
			}

			if (c == "L" || c == "C") && strSplit[i-1] == "X" {
				total -= 20
			}

			if (c == "D" || c == "M") && strSplit[i-1] == "C" {
				total -= 200
			}
		}

		total += symbolMap[c]
	}

	return total
}

func run_romanToInt() {
	fmt.Println(romanToInt("MCMXCIV"))
}

File: Easy/roman_to_integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    const symbolMap = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000};
    s = s.toUpperCase().split("");

    return s.map((c, i) => {
        if(Object.hasOwn(symbolMap, c)) {
            let total = symbolMap[c];
            if(i > 0) {
                if ((c == "V" || c == "X") && s[i-1] == "I") {
                    total -= 2;
                }

                if ((c == "L" || c == "C") && s[i-1] == "X") {
                    total -= 20;
                }

                if ((c == "D" || c == "M") && s[i-1] == "C") {
                    total -= 200;
                }
            }

            return total;
        }
    }).reduce((a, b) => Number(a) + Number(b), 0);
};

console.log(romanToInt("MCMXCIV"));